lib/checkstl.cpp

```diff
@@ -1340,19 +1340,24 @@ void CheckStl::stlBoundariesError(const Token *tok)
                 "container is not guaranteed. One should use operator!= instead to compare iterators.", CWE664, false);
 }
 
-static bool if_findCompare(const Token * const tokBack)
+static bool if_findCompare(const Token * const tokBack, bool stdStringLike)
 {
     const Token *tok = tokBack->astParent();
     if (!tok)
         return true;
-    if (tok->isComparisonOp())
+    if (tok->isComparisonOp()) {
+        if (stdStringLike) {
+            const Token * const tokOther = tok->astOperand1() == tokBack ? tok->astOperand2(): tok->astOperand1();
```

Comment:
hmm.. tokBack can't be NULL here, can it? if tokBack is NULL there would be null pointer dereference at line 1345.

Reply:
not sure.. would it make sense to use `Token::astSibling` here?

Reply:
I meant `const Token * const tokOther = (tok->astOperand1() == tokBack) ? tok->astOperand2(): tok->astOperand1();`. I'll clarify.

Reply:
```would it make sense to use Token::astSibling here?```
I'd say yes. Thanks for the suggestion.